### PR TITLE
Add conversation cache

### PR DESCRIPTION
**Pull request by istvan-kreisz**

### CODE REVIEW

File: src/lib/useMessageCache.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { StreamingStepInput } from "./types";
+
+function useMessageCache(
+  setConversationId: React.Dispatch<React.SetStateAction<number>>,
```
Comment (Henry-Pulver) [inline src/lib/useMessageCache.ts:4]: Don't think this is _important_, but my pycharm linter seems to think we should be importing `React` in this file to use it :shrug:

File: src/lib/useMessageCache.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const cachedConversation: {
+      conversationId: number;
+      messages: StreamingStepInput[];
+      updated: Date;
+    } = JSON.parse(cachedConversationString);
+    cachedConversation.updated = new Date(cachedConversation.updated);
+
+    if (!cachedConversation) return;
+
+    console.log(!isExpired(cachedConversation.updated));
```
Comment (Henry-Pulver) [inline src/lib/useMessageCache.ts:32]: Cut?

File: src/lib/useMessageCache.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { StreamingStepInput } from "./types";
+
+function useMessageCache(
```
Comment (Henry-Pulver) [inline src/lib/useMessageCache.ts:3]: Why not `export default` here?
```suggestion
export default function useMessageCache(
```

**Review by Henry-Pulver — APPROVED** [review]
V minor tweaks suggested!

File: src/lib/useMessageCache.ts (hunk 1 above)
Comment (istvan-kreisz) [reply src/lib/useMessageCache.ts:4]: okay mine didn't say anything, but I'll add it then

File: src/lib/useMessageCache.ts (hunk 2 above)
Comment (istvan-kreisz) [reply src/lib/useMessageCache.ts:32]: yeah good call

File: src/lib/useMessageCache.ts (hunk 3 above)
Comment (istvan-kreisz) [reply src/lib/useMessageCache.ts:3]: yeah will do

**Pull request merged**